packages/mongodb-memory-server-core/src/util/MongoBinaryDownloadUrl.ts

```diff
@@ -14,10 +13,10 @@ import {
 const log = debug('MongoMS:MongoBinaryDownloadUrl');
 
 export interface MongoBinaryDownloadUrlOpts {
-  version: NonNullable<BaseDryMongoBinaryOptions['version']>;
+  version: string;
   platform: string;
-  arch: NonNullable<BaseDryMongoBinaryOptions['arch']>;
-  os?: BaseDryMongoBinaryOptions['os'];
+  arch: string;
+  os?: AnyOS;
```

Comment:
why was this changed? the options were made this way to have a single place define the type across the project

Reply:
The problem is that there is a circular dependency between `DryMongoBinary.ts` and `MongoBinaryDownloadUrl.ts`.
Since  DryMongoBinary uses the getArchiveName from MongoBinaryDownloadUrl.

Reply:
thanks for pointing this out, i will think about refactoring these types at some point